pkgs/darkest-dungeon/default.nix

```diff
@@ -0,0 +1,100 @@
+{
+  lib,
+  requireFile,
+  makeDesktopItem,
+  copyDesktopItems,
+  autoPatchelfHook,
+  makeBinaryWrapper,
+  stdenvNoCC,
+  zip,
+  unzip,
+  SDL2,
+  libcxx,
+  libGL,
+  fmodex,
+  libmx,
+  libX11,
+  libXext,
+  libxcb,
+  libXau,
+  libXdmcp,
+}: let
+  commonDeps = [
+    SDL2
+    libcxx
+    libGL
+    fmodex
+    libmx
+    fmodex
+    libX11
+    libXext
+    libxcb
+    libXau
+    libXdmcp
+  ];
+in
+  stdenvNoCC.mkDerivation rec {
+    pname = "darkest-dungeon";
+    version = "24839_28859";
+
+    # The official installer must be provided by the end user.
+    src = requireFile {
+      name = "darkest_dungeon_${version}.sh";
+      url = "https://www.gog.com/game/darkest_dungeon";
+      sha256 = "2b235b43d8d6ab4b2a4b80b627ce2a9c31c1b4bee16fc6f9dda68c29cb5da99c";
+    };
+
+    nativeBuildInputs =
+      [
+        autoPatchelfHook
+        makeBinaryWrapper
+        unzip
+        zip
+        copyDesktopItems
+      ]
+      ++ commonDeps;
+
+    desktopItems = [
+      (makeDesktopItem {
+        name = pname;
+        desktopName = "Darkest Dungeon";
+        genericName = "Loosing your will to live; the game.";
+        exec = "darkest";
+        #icon = "CCicon";  # The proper icon needs to be set instead.
```

Comment:
Can probably be extracted from the sh in some way, otherwise you can add a `fetchurl` just like the other packages.

Reply:
Done. The desktop entry should now use the icon provided by the game.